bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/firmware/FirmwareBuilder.java

```diff
@@ -177,6 +180,21 @@ public FirmwareBuilder withMd5Hash(@Nullable String md5Hash) {
         return this;
     }
 
+    /**
+     * Additional restrictions can be applied on the firmware by providing
+     * {@link FirmwareRestriction}.
+     *
+     * @param customRestrictions a {@link FirmwareRestriction} for applying additional
+     *            restrictions on the firmware (not null)
+     * @return the updated builder
+     * @throws IllegalArgumentException if the given function is null
+     */
+    public FirmwareBuilder withCustomRestrictions(FirmwareRestriction customRestrictions) {
```

Comment:
rename also here. The fact that multiple checks could possibly be enclosed in the `FirmwareRestriction` instance is not crucial to the naming.
please review all other places.

Reply:
Sounds good, I will update the PR accordingly.